domain/src/main/java/life/qbic/domain/usermanagement/UserDomainService.java

```diff
@@ -34,15 +36,51 @@ public UserDomainService(UserRepository userRepository) {
    */
   public void createUser(String fullName, String email, char[] rawPassword) throws UserException {
     // First check, if a user with the provided email already exists
-    if (userRepository.findByEmail(email).isPresent()) {
-      throw new UserException("User with email address already exists.");
-    }
+    validateUserCredentials(fullName, email, rawPassword);
     var domainEventPublisher = DomainEventPublisher.instance();
```

Comment:
What you do here is a different behaviour then to lookup whether a user already exists. You could replace that with `isEmailInDatabase` but I do not think that it adds readability here.